### PR TITLE
Performance: Goods endpoints

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: goods/serializers.py [hunk 1]
```diff
@@ -349,15 +260,75 @@ class Meta:
...
+    def __init__(self, *args, **kwargs):
+        super(GoodSerializerExporterFullDetail, self).__init__(*args, **kwargs)
+        self.goods_query = GoodsQuery.objects.filter(good=self.instance).first()
+
+    def get_case_id(self, instance):
+        return str(self.goods_query.id) if self.goods_query else None
+
+    def get_documents(self, instance):
+        documents = GoodDocument.objects.filter(good=instance)
+        if documents.exists():
```
Comment (a-bek) [inline goods/serializers.py:304]: Checking if it exists isn't necessary, could just return None if the serialiser data is empty.

File: goods/serializers.py [hunk 2]
```diff
@@ -349,15 +260,75 @@ class Meta:
...
+    def get_query(self, instance):
+        from queries.goods_query.serializers import ExporterReadGoodQuerySerializer
+
+        if self.goods_query:
+            serializer = ExporterReadGoodQuerySerializer(
+                instance=self.goods_query,
+                context={"exporter_user": self.context.get("exporter_user"), "total_count": False},
+            )
+            return serializer.data
+        return None
```
Comment (a-bek) [inline goods/serializers.py:317]: `return None` is redundant

File: goods/serializers.py (hunk 1 above)
Comment (BenAndrew-Kainos) [reply goods/serializers.py:304]: done

File: goods/serializers.py (hunk 2 above)
Comment (BenAndrew-Kainos) [reply goods/serializers.py]: done

File: goods/views.py [hunk 3]
```diff
@@ -152,8 +153,43 @@ def get_queryset(self):
...
+        goods_queries = {
+            str(query["id"]): {"good": str(query["good_id"]), "exporter_user_notification_count": 0}
+            for query in goods_queries
+        }
+
+        notifications = (
+            ExporterNotification.objects.filter(
+                user=self.request.user,
+                organisation_id=get_request_user_organisation_id(self.request),
+                case__id__in=goods_queries.keys(),
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply goods/views.py]: done

**Pull request merged**